cmd/end2endtest/zkweighted.go

```diff
@@ -163,8 +167,58 @@ func (t *E2EAnonElectionTempSIKs) Run() error {
 		return err
 	}
 
+	// the half of accounts should not have valid sik, due they are using temporal sik
+	halfAccts := len(votes) / 2
+	accs, err := removedSiks(t.api, votes)
+	if err != nil {
+		return err
+	}
+
+	if int(accs) != halfAccts {
+		return fmt.Errorf("unexpected number of accounts without valid SIK, got %d want %d", accs, halfAccts)
+	}
+
 	log.Infof("election %s status is RESULTS", t.election.ElectionID.String())
 	log.Infof("election results: %v", elres.Results)
 
 	return nil
 }
+
+// removedSiks get the number of accounts without a valid SIK
+func removedSiks(api *apiclient.HTTPclient, votes []*apiclient.VoteData) (accts int32, err error) {
+	wg := &sync.WaitGroup{}
+	errorChan := make(chan error)
+	var siksRemoved int32
+
+	for _, v := range votes {
```

Comment:
mmh this spawns `len(votes)` concurrent goroutines, i'm worried about a memory spike. why not limit the number of parallel processes?